### PR TITLE
import personalizsation information from etsy as an order note for ea…

### PR DESCRIPTION
**Pull request by m-siebert**
…ch variation

### CODE REVIEW

File: src/Services/Order/OrderCreateService.php [hunk 1]
```diff
@@ -333,7 +338,7 @@ private function getOrderItems(array $data)
 					],
 					'properties'      => [
 						[
-							'typeId'    => OrderPropertyType::SELLER_ACCOUNT,
```
Comment (jschaakplenty) [inline src/Services/Order/OrderCreateService.php:336]: might this have an impact on aftersales processes?

File: src/Services/Order/OrderCreateService.php [hunk 2]
```diff
@@ -454,15 +459,51 @@ private function getVariationById(int $variationId)
...
+                            foreach ($order->orderItems as $orderItem) {
+                                try {
+                                    $orderItemExternalItemIdProperty = $orderItemPropertyRepo->findByOrderItemIdAndTypeId(
+                                        $orderItem->id,
+                                        OrderPropertyType::EXTERNAL_ITEM_ID
+                                    );
+                                    $orderItemExternalItemId = $orderItemExternalItemIdProperty->value;
+                                } catch (\Exception $e) {
+                                    continue;
+                                }
```
Comment (jschaakplenty) [inline src/Services/Order/OrderCreateService.php:490]:
```suggestion
                                $orderItemExternalItemId = $orderItem->property(OrderPropertyType::EXTERNAL_ITEM_ID);
```

this function returns the value if property exists or null if not.

File: src/Services/Order/OrderCreateService.php [hunk 3]
```diff
@@ -454,15 +459,51 @@ private function getVariationById(int $variationId)
...
+
+                                if ($orderItemExternalItemId == (string)$transaction['listing_id']) {
+                                    $commentRepo->createComment([
+                                            'referenceType'       => Comment::REFERENCE_TYPE_ORDER,
+                                            'referenceValue'      => $orderId,
+                                            'isVisibleForContact' => true,
+                                            'text'                => '<b>' . $this->translator->trans('Etsy::order.personalizationMessage') . $orderItem->itemVariationId . ':</b><br><br>' . nl2br(html_entity_decode($attribute['formatted_value']))
+                                        ]);
+                                    break 3;
```
Comment (jschaakplenty) [inline src/Services/Order/OrderCreateService.php:499]:
```suggestion
                                    $comment = [
                                            'referenceType'       => Comment::REFERENCE_TYPE_ORDER,
                                            'referenceValue'      => $orderId,
                                            'isVisibleForContact' => true,
                                            'text'                => '<b>' . $this->translator->trans('Etsy::order.personalizationMessage') . $orderItem->itemVariationId . ':</b><br><br>' . nl2br(html_entity_decode($attribute['formatted_value']))
                                    ];
                                    
                                    $commentRepo->createComment(comment);
                                    break 3;
```

better style for debugging

**Review by jschaakplenty — CHANGES_REQUESTED** [review]

File: src/Services/Order/OrderCreateService.php (hunk 1 above)
Comment (m-siebert) [reply src/Services/Order/OrderCreateService.php:336]: I searched for `OrderPropertyType::SELLER_ACCOUNT` in the whole directory and can't find any other place where it is used

Comment (jschaakplenty) [reply src/Services/Order/OrderCreateService.php:336]: did you also search for 10? 😄

Comment (m-siebert) [reply src/Services/Order/OrderCreateService.php:336]: now yes 😄 but no results either

**Review by jschaakplenty — APPROVED** [review]

**Review by jschaakplenty — APPROVED** [review]

File: src/Services/Order/OrderCreateService.php [hunk 4]
```diff
@@ -17,6 +17,7 @@
 use Plenty\Modules\Order\Contracts\OrderRepositoryContract;
 use Plenty\Modules\Order\Models\Order;
 use Plenty\Modules\Order\Models\OrderItemType;
+use Plenty\Modules\Order\Property\Contracts\OrderItemPropertyRepositoryContract;
```
Comment (jschaakplenty) [inline src/Services/Order/OrderCreateService.php:20]: can also be removed

**Review by jschaakplenty — CHANGES_REQUESTED** [review]

**Review by khoffmeister — APPROVED** [review]

**Pull request merged**